cloud/wfm/src/executor/ExecutorImpl.ts

```diff
@@ -0,0 +1,37 @@
+import * as Promise from 'bluebird';
+import {Process} from '../process';
+import {ProcessInstance} from '../process-instance';
+import {Task, TaskEventData, TaskStatus} from '../task';
+import {Executor} from './index';
+import {InstanceRepository} from './InstanceRepository';
+
+export {InstanceRepository};
+export default class ExecutorImpl implements Executor {
+  public instance: ProcessInstance;
+  constructor(public process: Process, public instanceRepository: InstanceRepository) {
+    this.instance = this.process.createInstance();
+  }
+  public start() {
+    this.runCurrentTask();
+  }
+  protected onTaskDone(e: TaskEventData<Task>) {
```

Comment:
Can we have this and possibly other events as part of the TaskEventListener interface? 
Thats good pattern when users will want to listen to events. Executor can both issue events and also provide default listeners?


onTaskDone sounds strange. Are you ok with `onTaskFinished`?